onnxruntime/core/optimizer/gelu_fusion.cc

```diff
@@ -70,10 +70,13 @@ static bool IsSupportedDataType(const Node& node) {
 Status GeluFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level) const {
   GraphViewer graph_viewer(graph);
   const auto& node_topology_list = graph_viewer.GetNodesInTopologicalOrder();
-  std::deque<onnxruntime::NodeIndex> removed_nodes;
 
   for (auto node_index : node_topology_list) {
-    auto& div = *graph.GetNode(node_index);
+    auto* p_div = graph.GetNode(node_index);
+    if (p_div == nullptr)
```

Comment:
Does this nullptr check need for the remaining erf_node, add_node and mul_node?

Reply:
I don't believe so. We only remove them if they're in a chain, so there would be no other edge to those nodes to be able to access them, and when we hit them via their node_index this check will kick in.

Note that p_div is a bit misleading - this is any node type until after line 82 passes.

---
In reply to: [337235788](https://github.com/microsoft/onnxruntime/pull/2213#discussion_r337235788) [](ancestors = 337235788)

Reply:
I don't understand why we don't need to check null for erf_node. What if div is the edge node?

---
In reply to: [337263635](https://github.com/microsoft/onnxruntime/pull/2213#discussion_r337263635) [](ancestors = 337263635,337235788)

Reply:
Scott synced with me offline. It is good.

---
In reply to: [337266790](https://github.com/microsoft/onnxruntime/pull/2213#discussion_r337266790) [](ancestors = 337266790,337263635,337235788)

Reply:
We navigate to downstream nodes via output edges. If we did a previous fusion we have removed all the nodes that were fused and their edges. They will still have a stale entry in node_topology_list, which is what this check is handling. 

---
In reply to: [337266790](https://github.com/microsoft/onnxruntime/pull/2213#discussion_r337266790) [](ancestors = 337266790,337263635,337235788)